dynamax/nonlinear_gaussian_ssm/inference_ekf.py

```diff
@@ -103,6 +104,9 @@ def extended_kalman_filter(
 
     """
     num_timesteps = len(emissions)
+    if outputs is None:
```

Comment:
Actually, rather than having  a string name for all 8 combinations, maybe it would be easier to have 3 boolean flags,
which we set to default values to match old behavior
```
store_means: Bool = True 
store_covs: Bool = True
store_logliks: Bool = False
```
Then the understanding is that the final `FilteredPosterior` object is either the final value or the history of values,
depending on the status of these flags.

Reply:
If we're going with this API, I propose to also add `store_intermediate_predictions` to obtain intermediate predicted values (for the case of online training/testing of methods)